src/test/compile-fail/catch-opt-init.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(catch_expr)]
+
+pub fn main() {
+    let cfg_res;
+    let _: Result<(), ()> = do catch {
+        Err(())?;
+        cfg_res = 5;
+        Ok(())
```

Comment:
can you modify the test to demonstrate that `cfg_res` **is** usable **within* the `do catch`, if not outside?

Also, add a (run-pass) test like so, where `cfg_res` is initialized *before* the `?`:

```rust
pub fn main() {
    let cfg_res;
    let _: Result<(), ()> = do catch {
        cfg_res = 5;
        Err(())?;
        Ok(())
    };
    use(cfg_res); // OK
}
```